internal/server/pool_collector.go

```diff
@@ -0,0 +1,100 @@
+package server
+
+import "github.com/prometheus/client_golang/prometheus"
+
+type poolCollector struct {
+	p *Pool
+
+	maxOpenConnections *prometheus.Desc
+
+	openConnections  *prometheus.Desc
+	inUseConnections *prometheus.Desc
+	idleConnections  *prometheus.Desc
+
+	waitCount         *prometheus.Desc
+	waitDuration      *prometheus.Desc
+	maxIdleClosed     *prometheus.Desc
+	maxIdleTimeClosed *prometheus.Desc
+	maxLifetimeClosed *prometheus.Desc
+}
+
+func newPoolCollector(p *Pool, destHost, destDB string) *poolCollector {
+	fqName := func(name string) string {
+		return namespace + "_" + name
+	}
+	return &poolCollector{
+		p: p,
+		maxOpenConnections: prometheus.NewDesc(
+			fqName("max_open_connections"),
+			"Maximum number of open connections to the database.",
+			nil, prometheus.Labels{"dest_host": destHost, "dest_db": destDB},
+		),
+		openConnections: prometheus.NewDesc(
+			fqName("open_connections"),
+			"The number of established connections both in use and idle.",
+			nil, prometheus.Labels{"dest_host": destHost, "dest_db": destDB},
+		),
+		inUseConnections: prometheus.NewDesc(
+			fqName("in_use_connections"),
+			"The number of connections currently in use.",
+			nil, prometheus.Labels{"dest_host": destHost, "dest_db": destDB},
+		),
+		idleConnections: prometheus.NewDesc(
+			fqName("idle_connections"),
+			"The number of idle connections.",
+			nil, prometheus.Labels{"dest_host": destHost, "dest_db": destDB},
+		),
+		waitCount: prometheus.NewDesc(
+			fqName("wait_count_total"),
+			"The total number of connections waited for.",
+			nil, prometheus.Labels{"dest_host": destHost, "dest_db": destDB},
+		),
+		waitDuration: prometheus.NewDesc(
+			fqName("wait_duration_seconds_total"),
+			"The total time blocked waiting for a new connection.",
```

Comment:
I'm of the opinion that it's never too much information in these descriptions, even if the name is already clear:
```suggestion
			"The total time (in seconds) blocked waiting for a new connection.",
```